website/static/js/pages/sharing-page.js

```diff
@@ -24,6 +29,48 @@ $('body').on('nodeLoad', function(event, data) {
     }
 });
 
+$('.filters').filters({
+    items: ['.contrib', '.admin'],
+    callback: function (filtered, empty) {
+        if ($.inArray('.contrib', empty) > -1) {
+            $('#noContributors').show();
+        }
+        else {
+            $('#noContributors').hide();
+        }
+        if ($.inArray('.admin', empty) > -1) {
+            $('#noAdminContribs').show();
+        }
+        else {
+            $('#noAdminContribs').hide();
+        }
+        var isFiltered = $.inArray('.contrib', filtered) > -1;
+        ko.contextFor($('#contributors').get(0)).$parent.sortable(isFiltered);
+    },
+    groups: {
+        permissionFilter: {
+            filter: '.permission-filter',
+            type: 'text',
+            buttons: {
+                admins: 'Administrator',
+                write: 'Read + Write',
+                read: 'Read'
+                }
+        },
+        visibleFilter: {
+            filter: '.visible-filter',
+            type: 'checkbox',
+            buttons: {
+                visible: true,
+                notVisible: false
```

Comment:
Oh nevermind, I thought this was controlling button visibility. But it's actually the names.